src/map/mail.cpp

```diff
@@ -99,15 +99,18 @@ int mail_removeitem(map_session_data *sd, short flag, int idx, int amount)
 	return 1;
 }
 
-bool mail_removezeny( map_session_data *sd, bool flag ){
+bool mail_removezeny(map_session_data *sd, bool flag) {
 	nullpo_retr( false, sd );
 
 	if( sd->mail.zeny > 0 ){
 		//Zeny send
 		if( flag ){
-			if( pc_payzeny( sd, sd->mail.zeny + sd->mail.zeny * battle_config.mail_zeny_fee / 100, LOG_TYPE_MAIL, NULL ) ){
+			if (pc_payzeny(sd, sd->mail.zeny + sd->mail.zeny * battle_config.mail_zeny_fee / 100, LOG_TYPE_NONE, NULL)) {
 				return false;
 			}
+			// we don't know the target char_id, just the name. So, let's just use 0.
+			// it's not that important anyways, as the recv log will show it as well.
+			log_zeny(*sd, LOG_TYPE_MAIL, 0, -sd->mail.zeny);
```

Comment:
I believe there is a potential workaround there. After 20150513 (approximate date), it is required to validate the destination character name before sending the mail. The information is received from the clif_Mail_Receiver_Ack packet. Adding a new field such as sd->mail.target_charid = char_id; in that packet and then using that value for the log_zeny would probably be more than enough. Just gotta make sure to reset it in mail_clear too.